testing/src/main/java/org/oppia/testing/FakeCrashLogger.kt

```diff
@@ -0,0 +1,30 @@
+package org.oppia.testing
+
+import java.lang.Exception
+import org.oppia.testing.TestLogReportingModule.Companion.exceptionList
+import org.oppia.util.firebase.CrashLogger
+
+/**
+ * A test specific fake crash logger that adds the logged exception to a list of exceptions.
+ */
+class FakeCrashLogger : CrashLogger {
+
+  override fun logException(exception: Exception) {
+    exceptionList.add(exception)
+  }
+
+  /** This is used to get the most recent exception present in the list of exceptions. */
+  fun getMostRecentException(): Exception {
+    val size = exceptionList.size
+    return if (size > 0) {
+      exceptionList[size - 1]
+    } else {
+      exceptionList[size]
```

Comment:
This will return an ArrayIndexOutOfBoundsException , your size is 0 so it will try to return the 0th element which doesn't exist. You should throw an error here not send exceptionList[size]

Reply:
Yes, my bad :( 
Modified it to throw `NullPointerException("No element present")`, will it be good enough ?

Reply:
This looks fine